relayer/src/chain/cosmos/encode.rs

```diff
@@ -0,0 +1,179 @@
+use bech32::{ToBase32, Variant};
+use core::str::FromStr;
+use ibc::core::ics24_host::identifier::ChainId;
+use ibc_proto::cosmos::tx::v1beta1::mode_info::{Single, Sum};
+use ibc_proto::cosmos::tx::v1beta1::{AuthInfo, Fee, ModeInfo, SignDoc, SignerInfo, TxBody, TxRaw};
+use ibc_proto::google::protobuf::Any;
+use tendermint::account::Id as AccountId;
+
+use crate::chain::cosmos::types::account::{Account, AccountNumber, AccountSequence};
+use crate::chain::cosmos::types::tx::SignedTx;
+use crate::config::types::Memo;
+use crate::config::AddressType;
+use crate::config::ChainConfig;
+use crate::error::Error;
+use crate::keyring::{sign_message, KeyEntry};
+
+pub fn sign_and_encode_tx(
```

Comment:
This is refactored from:

https://github.com/informalsystems/ibc-rs/blob/3e34b4b916eb91f680412e5fdd3ea045a99526d2/relayer/src/chain/cosmos.rs#L357-L369